tests/sentry/workflow_engine/endpoints/test_organization_detector_index.py

```diff
@@ -496,6 +496,56 @@ def test_query_by_assignee_invalid_user(self) -> None:
         )
         assert len(response.data) == 0
 
+    def test_query_by_project_owner_user(self) -> None:
+        new_project = self.create_project(organization=self.organization)
+        detector = self.create_detector(
+            project_id=new_project.id, name="Test Detector", type=MetricIssue.slug
+        )
+
+        owner = self.create_user()
+        self.create_member(
+            user=owner,
+            role="owner",
+            organization=self.organization,
+        )
+        self.login_as(user=owner)
+
+        # Verify that the owner can see detectors across all projects
+        response = self.get_success_response(
+            self.organization.slug,
+            qs_params={"project": new_project.id},
+            status_code=200,
+        )
+        assert {d["name"] for d in response.data} == {detector.name}
+
+    def test_query_by_id_owner_user(self) -> None:
+        self.detector = self.create_detector(
+            project_id=self.project.id,
+            name="Detector 1",
+            type=MetricIssue.slug,
+        )
+        self.detector_2 = self.create_detector(
+            project_id=self.project.id,
+            name="Detector 2",
+            type=MetricIssue.slug,
+        )
+
+        owner = self.create_user()
+        self.create_member(
+            user=owner,
+            role="owner",
+            organization=self.organization,
+        )
+        self.login_as(user=owner)
+
+        # Verify that the owner can see detectors across all projects
```

Comment:
The response shows detectors in one project, though the test does properly return the detectors by name - not sure if this was a copy/paste comment thing

Reply:
updated the comment, this is testing the same idea that the owner can access detectors for projects that they are not members of